[WIP] MBS-12012: Allow "dupe" URL rels with diff dates

reosarevok opened the pull request:
### Fix MBS-12012 / Implement MBS-12013

2026: Updating this for the new external link code.

# Problem
We have no way to indicate that a link stopped working, then came back. See MBS-13605 for a real life example.

# Solution
This changes link validation to only consider links duplicates if they have the same date period.

It also allows adding (and removing) additional relationships even for autoselected links, as long as the first one has dates or ended set (MBS-12013).

# AI usage
None

# Testing
Manually:
* Add two "official homepage" links, one with dates and one without, and see they can be submitted and edited.
* Check that the "Add another relationship" button appears for autoselected links but only after adding dates to them (MBS-12013)
* Check that validation accepts multiple relationships of the same type for a link, even for autoselected ones, as long as they are not duplicates.
* Check that complete duplicates (same dates) are still detected as such.
* Check that different relationships with different dates but the same type are no longer dropped when merging links.

reosarevok commented:
Cherry-picked 068b6c7a821ecf7432eeb7db546cac69b9dcf771 into its own PR since it's blocking users from editing so we should fix that soonish: https://github.com/metabrainz/musicbrainz-server/pull/2322

yvanzo commented:
@reosarevok: Can you please rebase it on current `master` to see if CI tests pass again?

reosarevok commented:
I can, but I doubt they will, as I said above I think that's because of it hitting MBS-12032. You've played with this code more, do you have any suggestions on how to fix that bug?

yvanzo commented:
I probably have to review #2322 in the first place?

reosarevok commented:
Well, that's been merged but you can take a look :) The bug existed before that though.

atj commented:
Please ping me when this ends up on beta ([related irc chat](https://chatlogs.metabrainz.org/libera/metabrainz/2023-05-05/?msg=5169490&page=1)). Thanks!

reosarevok committed:
[WIP] MBS-12012: Allow "dupe" URL rels with diff dates

reosarevok commented:
Changing the implementation of MBS-12012 so it applies to the new external links code (the other ticket here is now fixed by said code).

reosarevok committed:
Drop dupe rels merge check

I cannot see a good reason to keep this - any relationships
caught by this would generally be duplicates and
as such be generating a duplicate error, so we should
never hit this to begin with since links with rels that have
errors cannot be merged.
If anything, I could see us dropping any rels with
no type set?
